### PR TITLE
Fix reply to a review does not load after deletion

### LINKED ISSUE #6447: A new developer reply to a review does not load after deletion
### Describe the problem and steps to reproduce it:

- Sign in as a developer
- Go to one of your add-ons
- Go to the review listing
- Reply to a review
- Click on the timestamp 
- Delete the reply
- Reply to the review again
- Click on the timestamp again

### What happened?

Nothing happens

### What did you expect to happen?

You should see the permalink screen of the review reply

### Anything else we should know?

If you refresh the page, you can see the permalink screen as expected.

### PR DESCRIPTION
**Pull request by xu3u4**
Fixes #6447

#### Description
Clicking the timestamp permalink of a reply, it will become a featured review.
Delete it and then reply to the same review, the featured review should not be not found.

### CODE REVIEW

File: src/amo/components/FeaturedAddonReview/index.js [hunk 1]
```diff
@@ -89,6 +89,7 @@ export class FeaturedAddonReviewBase extends React.Component<InternalProps> {
       : null;
 
     const featuredReviewCard =
+      !featuredReview &&
       errorHandler.hasError() &&
       errorHandler.capturedError.responseStatusCode === 404 ? (
```
Comment (xu3u4) [inline src/amo/components/FeaturedAddonReview/index.js]: I may need some hint here.
`featuredReview` is loaded but the error about the not found reply is still there.
```
errors: {
  src/amo/components/FeaturedAddonReview/index.js-564315: {
    code: "ERROR_UNKNOWN",
    messages: ["Not found."],
    responseStatusCode: 404,
  }
}
```
I can't find a good place to clear that error...

**Comment by xu3u4** [comment]
@kumar303 
I see. Thank you.
Should I close it?

**Comment by kumar303** [comment]
Yeah, I think it will be better to close this one for now.

**Comment by kumar303** [comment]
Creating abstractions in Redux for common tasks like error handling is a little tricky. It's often cited as a weakness of Redux but I think it just requires different approaches. One approach is the reducer enhancer pattern. Here is an example I just found: https://medium.com/@mange_vibration/reducer-composition-with-higher-order-reducers-35c3977ed08f

**Pull request closed without merging**